awx_collection/plugins/modules/tower_project.py

```diff
@@ -229,10 +230,15 @@ def wait_for_project_update(module, last_request):
         start = time.time()
 
         # Invoke wait function
-        module.wait_on_url(
+        result_final = module.wait_on_url(
             url=result['json']['url'], object_name=module.get_item_name(last_request), object_type='Project Update', timeout=timeout, interval=interval
         )
 
+        # Set Changed to correct value depending on if hash changed Also output refspec comparision
+        module.json_output['changed'] = True
+        if result_final['json']['scm_revision'] == scm_revision_original:
```

Comment:
Food for thought..... I think on a technical level it can be more complicated that this. A project update can include a requirements.yml. if a new version of a collection/role was added the project hash itself may not change but that new version of the collection/role could have been downloaded changing Tower.

Reply:
Except in this case the proejct is https://github.com/ansible/test-playbooks without a collections/roles requirements files.

Reply:
At least considering The test, but In this case its not always going to work, but its was a requested feature to have it not changed when the scm does not change. I hadn't thought of the collections going in.

Reply:
Ok did test, Synced to 
https://github.com/sean-m-sullivan/verification_repo
and it synced with scm_revision 48214f29aa737c8ce093c339d6f4db0cbd74a59b
which matches   repo scm hash 48214f29aa737c8ce093c339d6f4db0cbd74a59b
and upon collection update got  48214f29aa737c8ce093c339d6f4db0cbd74a59b 
even though the tower_utilities version moved from 1.22 to 1.23

Reply:
Yes, but the API is vague in these cases. It is intended that a project update can pull in new versions of stuff from Galaxy (otherwise you would have to add new commits to rev it, yuck).

I've believed that the inevitable end-game is to track a checksum for the collections & roles folders specific to that project, and show that in the API.

In the absence of the API solving the problem, I'm not going to worry about the collection having false-negatives.

Reply:
Doesn't Ansible usually air on the side of caution and report false changes instead of false "OKs"? i.e. the command module returns changed by default.